usecases/case_usecase.go

```diff
@@ -465,11 +465,10 @@ func (uc *CaseUseCase) Snooze(ctx context.Context, req models.CaseSnoozeRequest)
 			previousSnooze = utils.Ptr(c.SnoozedUntil.Format(time.RFC3339))
 		}
 
-		if err := uc.repository.BoostCase(ctx, tx, req.CaseId, models.BoostUnsnoozed); err != nil {
```

Comment:
This reorders the side effects to happen before the boost, since one of the side effects is unboosting. A bit brittle, should I pass an option saying if unboost should happen?